app/services/custom_fields/custom_field_projects/bulk_create_service.rb

```diff
@@ -0,0 +1,125 @@
+# frozen_string_literal: true
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+module CustomFields
+  module CustomFieldProjects
+    class BulkCreateService < ::BaseServices::BaseCallable
+      def initialize(user:, projects:, custom_field:, include_sub_projects: false)
+        super()
+        @user = user
+        @projects = projects
+        @custom_field = custom_field
+        @include_sub_projects = include_sub_projects
+      end
+
+      def perform
+        service_call = validate_permissions
+        service_call = validate_contract(service_call, incoming_mapping_ids) if service_call.success?
+        service_call = perform_bulk_create(service_call) if service_call.success?
+
+        service_call
+      end
+
+      private
+
+      def validate_permissions(permission: :select_custom_fields)
+        return ServiceResult.failure(errors: I18n.t(:label_not_found)) if incoming_projects.empty?
+
+        if @user.allowed_in_project?(permission, incoming_projects)
+          ServiceResult.success
+        else
+          ServiceResult.failure(errors: I18n.t("activerecord.errors.messages.error_unauthorized"))
+        end
+      end
+
+      def validate_contract(service_call, project_ids)
+        set_attributes_results = project_ids.map do |id|
+          set_attributes(project_id: id, custom_field_id: @custom_field.id)
+        end
+
+        if (failures = set_attributes_results.select(&:failure?)).any?
+          service_call.success = false
+          service_call.errors = failures.map(&:errors)
+        else
+          service_call.result = set_attributes_results.map(&:result)
+        end
```

Comment:
🟡 Sorry if I'm a bit direct here, but I would treat a `ServiceResult` as a finite object and would rather create a new one and see every `ServiceResult` as immutable. That would also play nicely when using `bind` or `>>`. You would not need to have the `service_call` parameter at all for this function, because this function would only be called after the one before succeeds.